src/autocomplete/Autocompleter.js

```diff
@@ -52,28 +52,30 @@ export async function getCompletions(query: string, selection: SelectionRange, f
      otherwise, we run into a condition where new completions are displayed
      while the user is interacting with the list, which makes it difficult
      to predict whether an action will actually do what is intended
-
-     It ends up containing a list of Q promise states, which are objects with
-     state (== "fulfilled" || "rejected") and value. */
-    const completionsList = await Q.allSettled(
-        PROVIDERS.map(provider => {
-            return Promise.resolve(provider.getCompletions(query, selection, force))
-                .timeout(PROVIDER_COMPLETION_TIMEOUT);
-        }),
+    */
+    const completionsList = await Promise.all(
+        // Array of inspections of promises that might timeout. Instead of allowing a
+        // single timeout to reject the Promise.all, reflect each one and once they've all
+        // settled, filter for the fulfilled ones
+        PROVIDERS.map((provider) => {
+            // Convert to bluebird promise so that we can do a timeout
+            const p = Promise.resolve(provider.getCompletions(query, selection, force));
+            return p.timeout(PROVIDER_COMPLETION_TIMEOUT);
+        }).map((p) => p.reflect()),
```

Comment:
do you need a separate `map` here? why not just tag the `reflect` onto the `p.timeout` line?